src/ifaddrs.rs

```diff
@@ -0,0 +1,142 @@
+//! Query network interface addresses
+//!
+//! Uses the Linux and/or BSD specific function `getifaddrs` to query the list
+//! of interfaces and their associated addresses.
+
+use std::ffi;
+use std::iter::Iterator;
+use std::mem;
+use std::option::Option;
+
+use libc;
+
+use {Result, Errno};
+use sys::socket::SockAddr;
+use net::if_::*;
+
+/// Describes a single address for an interface as returned by `getifaddrs`.
+pub struct InterfaceAddress {
```

Comment:
Can we implement or derive `Debug`/`Display` for this.  Also, I think deriving `Clone`, `PartialEq`, and `Hash` would be good.

Reply:
I agree with this as well, should definitely have `Clone`, `Copy`, `Debug`, `Eq`, `Hash`, and `PartialEq`. We shouldn't implement `Display` here and I don't think we should implement `Default` here either.

Reply:
In commit ef40c9c, I added `derive`s for `Clone`, `Eq`, `Hash`, and `PartialEq`, but not `Copy`. The struct uses `String`s, so I don't this that marker is applicable.

Reply:
Oh, and `Debug`. I wasn't quite sure how much or in what format to return from `fn fmt`. Also note that `SockAddr` doesn't implement `Debug`, but `Display`. Is this really consistent?